pkg/apply/apply.go

```diff
@@ -49,6 +53,30 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 		return errors.Wrapf(err, "could not retrieve existing %s", objDesc)
 	}
 
+	if obj.GetKind() == "Namespace" {
+		ns := &v1.Namespace{}
+		if err := kscheme.Scheme.Convert(existing, ns, nil); err != nil {
+			return errors.Wrapf(err, "could not convert namespace %s", objDesc)
+		}
+		if ns.Status.Phase == v1.NamespaceTerminating {
+			log.Printf("finalize Namespace %s", objDesc)
+			ns.Spec.Finalizers = []v1.FinalizerName{}
```

Comment:
Hang on a moment - you can't just go and delete all finalizers. You should only delete the finalizer for which you are responsible.

Reply:
This logic will only be triggered when user hit a problem with OVN after the migration and node reboot. At that time all the openshift API is not available (as OVN doesn't work), so the finalizer of `openshift-sdn` namespace cannot be removed. In such situation, the user cannot rollback, as the `openshift-sdn` is hung in the `Terminating`, new objects cannot be recreated. 
So either we ask user to manually invoke the `finalize` API before rolling back, or we add this logic in the code before recreating the `openshift-sdn` namespace.